tools/tools/nanobsd/dhcpd/common

```diff
@@ -137,102 +147,103 @@ WITHOUT_INSTALLLIB=true
+	mount -t nullfs -o noatime "${NANO_SRC}" "${NANO_WORLDDIR}"/usr/src
+	mount -t nullfs -o noatime "${NANO_PORTS}" "${NANO_WORLDDIR}"/usr/ports
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/packages \
+	    "${NANO_WORLDDIR}"/usr/ports/packages
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/distfiles \
+	    "${NANO_WORLDDIR}"/usr/ports/distfiles
+	mkdir -p "${NANO_WORLDDIR}"/dev
+	mount -t devfs devfs "${NANO_WORLDDIR}"/dev
+	mkdir -p "${NANO_WORLDDIR}"/usr/workdir
+	cp /etc/resolv.conf "${NANO_WORLDDIR}"/etc/resolv.conf
 	# OK, a little inefficient, but likely not enough to worry about.
 	CR ldconfig /lib /usr/lib /usr/local/lib
 	CR ldconfig -R
 	CR ldconfig -r
-# Improvement: Don't know why package-recursive don't works here
-	CR "env UNAME_p=${NANO_ARCH} TARGET=${NANO_ARCH} \
+# Improvement: Don't know why package-recursive don't work here
+	CR "env UNAME_p=${NANO_ARCH} TARGET=${NANO_TARGET} \
 	    TARGET_ARCH=${NANO_ARCH} PORTSDIR=${NANO_PORTS} make \
 	    __MAKE_CONF=${NANO_MAKE_CONF_BUILD} \
 	    WRKDIRPREFIX=/usr/workdir -C /usr/ports/$port_path \
 	    package-recursive BATCH=yes $* clean FORCE_PKG_REGISTER=t"
-	rm ${NANO_WORLDDIR}/etc/resolv.conf
-	rm -rf ${NANO_WORLDDIR}/usr/obj
-	rm -rf ${NANO_WORLDDIR}/usr/workdir
-	umount ${NANO_WORLDDIR}/dev
-	umount ${NANO_WORLDDIR}/usr/ports/packages
-	umount ${NANO_WORLDDIR}/usr/ports/distfiles
-	umount ${NANO_WORLDDIR}/usr/ports
-	umount ${NANO_WORLDDIR}/usr/src
+	rm "${NANO_WORLDDIR}"/etc/resolv.conf
+	rm -rf "${NANO_WORLDDIR}"/usr/obj
+	rm -rf "${NANO_WORLDDIR}"/usr/workdir
+	umount "${NANO_WORLDDIR}"/dev
+	umount "${NANO_WORLDDIR}"/usr/ports/packages
+	umount "${NANO_WORLDDIR}"/usr/ports/distfiles
+	umount "${NANO_WORLDDIR}"/usr/ports
+	umount "${NANO_WORLDDIR}"/usr/src
 	set +x
 }
 
 # Need to check if this function works with cross-compiling architecture!!!!
-# Recursive complex fonction: Generate one function for each ports
-add_port () {
-    local port_path=$1
-    local port=`echo $1 | sed -e 's/\//_/'`
-    shift
-    # Check if package already exist
-    # Need to:
-    # 1. check ARCH of this package!
-    # 2. Add a trap
-    cd ${NANO_PORTS}/${port_path}
-    PKG_NAME=`env PORTSDIR=${NANO_PORTS} make __MAKE_CONF=${NANO_MAKE_CONF_BUILD} package-name` 
-    if [ -f ${NANO_OBJ}/ports/packages/All/${PKG_NAME}.txz ]; then
-	# Pkg file found: Generate add_pkg_NAME function
-	eval "
-	    add_pkg_${port} () {
-	        do_add_pkg ${PKG_NAME}
-	    }
-            customize_cmd add_pkg_${port}
-            "
-    else
-	# No pkg file: Generate add_port_NAME function
-        eval "
-            add_port_${port} () {
-	        do_add_port ${port_path} $*
-	    }
-	    customize_cmd add_port_${port}
-	"
-	NANO_PACKAGE_ONLY=0
-    fi
+# Recursive complex function: Generate one function for each ports
+add_port()
+{
+	local port_path="$1"
+	local port="$(echo "$1" | sed -e 's/\//_/')"
+	shift
+	# Check if package already exist
+	# Need to:
+	# 1. check ARCH of this package!
```

Comment:
Investigate a little more... I remember `ARCH`, et al. is/was used in other (embedded) scripts.